### PR TITLE
fix: hide IRM chart on vaults with no live borrow exposure

### PR DESCRIPTION
**Pull request by kasperpawlowski**
## Summary
- Hide the IRM chart on vaults with no live borrow-side exposure. Previously it was shown whenever `interestRateModelAddress` was non-zero, so collateral-only / escrow vaults could display it.
- Extract the collateral-exposure predicate used by the "Collateral exposure" block into a shared helper in `entities/vault/collateral-exposure.ts` and reuse it to gate the IRM chart, so both stay in sync.
- A pair is considered live when its current liquidation LTV is non-zero **and** either `borrowLTV > 0` or the collateral vault still has outstanding supply — i.e. open interest can still accrue while the LTV ramps down.

## Test plan
- [x] `npx vitest run tests/entities/vault/collateral-exposure.test.ts` — 16 new cases
- [x] `npx vitest run` — full suite (342/342)
- [x] `npm run typecheck` — exit 0
- [x] Borrowable vault detail page — IRM chart still renders.
- [x] Collateral-only / escrow vault detail page — IRM chart is hidden.
- [x] Vault mid ramp-down with outstanding debt (`borrowLTV = 0` but open positions) — IRM chart still renders.
- [x] "Collateral exposure" block lists the same pairs as before (refactor has no behavioural change on the borrow block).

### CODE REVIEW

File: components/entities/vault/overview/VaultOverviewBlockIRM.vue [hunk 1]
```diff
@@ -47,10 +48,20 @@ const { getChartColors, isDark } = useThemeColors()
...
+// vaults that may still carry a non-zero interestRateModelAddress.
 const hasValidIRM = computed(() => {
-  return vault.interestRateModelAddress
+  const hasExposure = hasCollateralExposure(
+    vault,
+    addr => registryGet(addr)?.vault as Vault | SecuritizeVault | undefined,
+  )
+  return hasExposure
+    && vault.interestRateModelAddress
     && vault.interestRateModelAddress !== '0x0000000000000000000000000000000000000000'
```
[the remark this answers is not in the document]
Comment (Seranged) [reply components/entities/vault/overview/VaultOverviewBlockIRM.vue]: Fixed

File: entities/vault/collateral-exposure.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ *    still accrue interest even after `borrowLTV` has been set to zero mid
+ *    ramp-down.
+ */
+const isLiveExposure = (
+  ltv: VaultCollateralLTV,
+  collateral: Vault | SecuritizeVault,
+  nowSeconds?: bigint,
+): boolean => {
+  if (getCurrentLiquidationLTV(ltv, nowSeconds) <= 0n) return false
+  return ltv.borrowLTV > 0n || collateral.totalAssets > 0n
```
[the remark this answers is not in the document]
Comment (Seranged) [reply entities/vault/collateral-exposure.ts:48]: Added an inline comment to isLiveExposure (its intentional)

File: components/entities/vault/overview/VaultOverviewBlockIRM.vue [hunk 3]
```diff
@@ -47,10 +48,20 @@ const { getChartColors, isDark } = useThemeColors()
...
+// Only render the IRM chart for vaults that have live borrow-side exposure —
+// either currently borrowable, or still accruing interest on existing debt
+// while the liquidation LTV ramps down. This mirrors the visibility rule of
+// the "Collateral exposure" block and correctly excludes collateral-only
+// vaults that may still carry a non-zero interestRateModelAddress.
 const hasValidIRM = computed(() => {
-  return vault.interestRateModelAddress
+  const hasExposure = hasCollateralExposure(
+    vault,
+    addr => registryGet(addr)?.vault as Vault | SecuritizeVault | undefined,
```
[the remark this answers is not in the document]
Comment (Seranged) [reply components/entities/vault/overview/VaultOverviewBlockIRM.vue:61]: Agreed it's a duplicate, but at two call sites it's not worth the  abstraction yet.

File: entities/vault/collateral-exposure.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  targetTimestamp: bigint
+  rampDuration: bigint
+}
+
+/**
+ * Resolves a collateral vault by address. Returns `undefined` when the
+ * collateral is unknown (not yet loaded in the vault registry).
+ */
+export type CollateralVaultResolver =
+  (address: string) => Vault | SecuritizeVault | undefined
```
[the remark this answers is not in the document]
Comment (Seranged) [reply entities/vault/collateral-exposure.ts:22]: Intentionally left as string — the values come from on-chain data so there's no untrusted input risk at this boundary

**Review by Seranged — APPROVED** [review]
Verified manually on mainnet: IRM chart renders correctly on borrowable vaults and is absent on escrow/collateral-only vaults. 

hasCollateralExposure correctly gates on live borrow-side exposure rather than just a non-zero IRM address, and the test suite covers the ramp-down edge cases precisely. 

Follow-up fixes for late registry hydration, zeroAddress consistency, and the totalAssets approximation comment are included.
